### ISSUE TITLE
Support silent authentication

### ISSUE THREAD

**Issue by jornane** [eduvpn/vpn-user-portal#132]
In the OAuth flow as implemented, users are asked to "allow" or "reject" the app they just used. Many of my users feel this is an extranous step and should not be part of the login flow.

By using [silent authentication](https://auth0.com/docs/api-auth/tutorials/silent-authentication), user interaction in the portal is no longer needed, and the user is redirected back to the redirect_url immediately. This improves the user experience.

**Comment by fkooman** [comment]
> In the OAuth flow as implemented, users are asked to "allow" or "reject" the app they just used. Many of my users feel this is an extranous step and should not be part of the login flow.

Because we feel our way to secure software ;-)

> By using silent authentication, user interaction in the portal is no longer needed, and the user is redirected back to the redirect_url immediately. This improves the user experience.

Sure it will improve usability. Except it is not safe against other applications running on the user's device stealing the access/authorization codes without the user noticing it (in SSO scenario).

On iOS and Android this should be a no-brainer as there it can be made secure _without_ prompt by registering and linking a HTTPS URL for the application in the App Store / Play Store. However, our iOS/Android developers are unable to get this to work. We tried:

- iOS: https://github.com/eduvpn/ios/issues/23
- Android: https://github.com/eduvpn/android/issues/129

On Windows and macOS we can make this work only by putting the applications in the Windows Store and App store, but so far no work has been done on this.

To sum up: the approval is required to prevent other applications installed on the user's device from obtaining the authorization_code/access_token without the user's involvement.

You can copy/paste all clients in https://github.com/eduvpn/vpn-user-portal/blob/v2/src/OAuthClientInfo.php to `/etc/vpn-user-portal/config.php` under `consumerList` and set `require_approval` to `false`, but **at your own risk**.

**Comment by fkooman** [comment]
Various attempts at fixing this resulted in nothing. We spent considerable time trying to get this to work. It seems impossible to fix it reliably, so not much we can do here, unfortunately. In theory it can be solved on iOS, Android and macOS (new store version), but all attempts either didn't work, didn't work on all version of the OS (Android) or worked sometimes (iOS).
